### ISSUE TITLE
Pressure zone for geometries in multiple meshes

### ISSUE THREAD

**Issue by marcosvanella** [firemodels/fds#9599]
In the attached case there is a closed compartment defined from box geometries. When the roof lays exactly in the boundary between the two meshes in Z (GEOM box &GEOM lines uncommented) the inner pressure zone gets a value of 0. When the roof in placed one cell higher in the top mesh (GEOM box 1 cell taller lines uncommented), the inner pressure zone gets the correct value of 1.

```
&HEAD CHID='test', TITLE='Test Pressure zone for multi mesh.' /

&MULT ID='mult', DX=3., I_UPPER=0, DY=3., J_UPPER=0, DZ=1.5, K_UPPER=1 /
&MESH ID='mesh', IJK=30,30,15, XB=-1.5,1.5,-1.5,1.5,-1.5,0, MULT_ID='mult'/

&VENT MB='XMIN', SURF_ID='OPEN' /

! OBST box:
#&OBST XB=-0.6,-0.5,-0.6, 0.6,-1.5,0.0 / ! Low X wall.
#&OBST XB= 0.5, 0.6,-0.6, 0.6,-1.5,0.0 / ! Hi  X wall.
#&OBST XB=-0.5, 0.5,-0.6,-0.5,-1.5,0.0 / ! Low Y wall.
#&OBST XB=-0.5, 0.5, 0.5, 0.6,-1.5,0.0 / ! Hi  Y wall.
#&OBST XB=-0.5, 0.5,-0.5, 0.5, 0.0,0.1 / ! Roof.

! GEOM box:
&GEOM XB=-0.6,-0.5,-0.6, 0.6,-1.5,0.0 / ! Low X wall.
&GEOM XB= 0.5, 0.6,-0.6, 0.6,-1.5,0.0 / ! Hi  X wall.
&GEOM XB=-0.5, 0.5,-0.6,-0.5,-1.5,0.0 / ! Low Y wall.
&GEOM XB=-0.5, 0.5, 0.5, 0.6,-1.5,0.0 / ! Hi  Y wall.
&GEOM XB=-0.5, 0.5,-0.5, 0.5, 0.0,0.1 / ! Roof. 

! GEOM box 1 cell taller:
#&GEOM XB=-0.6,-0.5,-0.6, 0.6,-1.5,0.1 / ! Low X wall.
#&GEOM XB= 0.5, 0.6,-0.6, 0.6,-1.5,0.1 / ! Hi  X wall.
#&GEOM XB=-0.5, 0.5,-0.6,-0.5,-1.5,0.1 / ! Low Y wall.
#&GEOM XB=-0.5, 0.5, 0.5, 0.6,-1.5,0.1 / ! Hi  Y wall.
#&GEOM XB=-0.5, 0.5,-0.5, 0.5, 0.1,0.2 / ! Roof.

&SLCF PBY=0., QUANTITY='PRESSURE ZONE', CELL_CENTERED=T/

&TAIL/

```

**Comment by marcosvanella** [comment]
Adding Isaac @idpark0129 to this thread.

**Comment by mcgratta** [comment]
I figured out what's wrong, but have no solution as yet. The problem is this---the uppermost grid cell in mesh 1, just below the ceiling, should not take the PRESSURE_ZONE value from mesh 2 because the interface between mesh 1 and 2 at that point is not `BOUNDARY_TYPE=INTERPOLATED_BOUNDARY`. It is not because there is a solid wall there. FDS knows this because one of the adjacent cells is SOLID. However, in the IBM case, mesh 1 assumes that the room ceiling is an INTERPOLATED boundary. It is not. It is a solid boundary as far as mesh 1 is concerned.

We need the IBM case to NOT set `BOUNDARY_TYPE=INTERPOLATED_BOUNDARY` at the top of mesh 1 inside the room.

**Comment by marcosvanella** [comment]
I see. The upper-most faces of mesh 1 have FCVAR(I,J,KBAR,IBM_FGSC,KAXIS)==IBM_SOLID, i.e. are defined as solid (completely inside or in the geometry boundary). We can make use of this info.

**Comment by mcgratta** [comment]
Every `EXTERNAL_WALL_CELL` has a unique `BOUNDARY_TYPE`. In your case, it is assumed that the boundary is `INTERPOLATED` because it sees that `SOLID(IC)=F` for the cells on either side. We need to fix this. Even though this boundary is an interface between two meshes, it is not an `INTERPOLATED_BOUNDARY` because gas cannot flow through it.

**Comment by marcosvanella** [comment]
We have a routine where we block external wall cells that are inside the geometry, BLOCK_IBM_SOLID_EXTWALLCELLS in geom.f90. OPEN and SOLID boundary types as set to NULL_BOUNDARY (no need for them if inside the geometry). We can add the INTERPOLATED there.

**Comment by mcgratta** [comment]
For the ceiling of the room, `WC%BOUNDARY_TYPE=SOLID_BOUNDARY`.

>

**Comment by marcosvanella** [comment]
This issue is addressed in PR#9605.
